### PR TITLE
feat: enable C6 in gatekeeping state (FPLA-948)

### PR DESCRIPTION
**Pull request by joez-f**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-948

### Change description ###

Enable C6 in gatekeeping state, and give the appropriate permissions to gatekeeper role so that they can complete the event in this state.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-definition/CaseEvent/Gatekeeping.json [hunk 1]
```diff
@@ -118,13 +118,29 @@
     "ShowEventNotes": "Y",
     "EndButtonLabel": "Save and continue"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "ID": "createNoticeOfProceedingsGatekeeping",
+    "Name": "Create notice of proceedings",
+    "Description": "Create notice of proceedings",
+    "DisplayOrder": 9,
```
Comment (Gogetter) [inline ccd-definition/CaseEvent/Gatekeeping.json]: is the display order correct?

File: ccd-definition/CaseEventToComplexTypes/notice-of-proceedings/NoticeOfProceedingsGatekeeping.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "NoticeOfProceedings",
+    "CaseEventID": "createNoticeOfProceedingsGatekeeping",
+    "CaseFieldID": "noticeOfProceedings",
+    "ListElementCode": "judgeAndLegalAdvisor.judgeTitle",
+    "EventElementLabel": "Judge or magistrate's title",
+    "FieldDisplayOrder": 1,
+    "DisplayContext": "OPTIONAL"
+  },
```
Comment (Gogetter) [inline ccd-definition/CaseEventToComplexTypes/notice-of-proceedings/NoticeOfProceedingsGatekeeping.json]: do we need _HintText_ for this?

Comment (joez-f) [reply ccd-definition/CaseEventToComplexTypes/notice-of-proceedings/NoticeOfProceedingsGatekeeping.json]: No, it's a radio list

File: ccd-definition/CaseEvent/Gatekeeping.json (hunk 1 above)
Comment (joez-f) [reply ccd-definition/CaseEvent/Gatekeeping.json]: To the best of my knowledge yes -  I don't think the display order for events has been properly revised. Not too important imo, probably needs a discussion later once we have all events in system? easy to swap display orders

Comment (joez-f) [reply ccd-definition/CaseEvent/Gatekeeping.json]: Screenshot 2019-11-25 at 12 49 33
Screenshot 2019-11-25 at 12 49 10
You can see here submitted vs gatekeeping event list. Fairly inconsistent! We should put it in a separate story later to make these consistent, not scope of this story to other fix display orders

File: ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json [hunk 3]
```diff
@@ -6,13 +6,27 @@
     "UserRole": "caseworker-publiclaw-gatekeeper",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseEventID": "createNoticeOfProceedingsGatekeeping",
```
Comment (akomisarek) [inline ccd-definition/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json]: We could consider releasing it finally if we are implemented the end time finally.

**Pull request merged**